### PR TITLE
recursively set indentations for child arrays

### PR DESCRIPTION
**Pull request by hxss**
PR for #163 
I was use auto generated complex arrays with personal data for tests, so I cant publish it.

### CODE REVIEW

File: test/Generator/ValueGeneratorTest.php [hunk 1]
```diff
@@ -225,6 +225,87 @@ public function complexArray()
         return $this->generateArrayData($longOutput, $value);
     }
 
+    /**
+     * Data provider for testPropertyDefaultValueCanHandleComplexArrayWCustomIndentOfTypes test
+     *
+     * @return array
+     */
+    public function complexArrayWCustomIndent()
```
Comment (Ocramius) [inline test/Generator/ValueGeneratorTest.php]: Please use `: array`

File: test/Generator/ValueGeneratorTest.php [hunk 2]
```diff
@@ -225,6 +225,87 @@ public function complexArray()
         return $this->generateArrayData($longOutput, $value);
     }
 
+    /**
+     * Data provider for testPropertyDefaultValueCanHandleComplexArrayWCustomIndentOfTypes test
+     *
+     * @return array
```
Comment (Ocramius) [inline test/Generator/ValueGeneratorTest.php]: Can be dropped

File: test/Generator/ValueGeneratorTest.php [hunk 3]
```diff
@@ -338,6 +419,23 @@ public function testPropertyDefaultValueCanHandleComplexArrayOfTypes($type, arra
...
     }
 
+    /**
+     * @dataProvider complexArrayWCustomIndent
+     *
+     * @param string $type
+     * @param array $value
+     * @param string $expected
+     */
+    public function testPropertyDefaultValueCanHandleComplexArrayWCustomIndentOfTypes($type, array $value, $expected)
```
Comment (Ocramius) [inline test/Generator/ValueGeneratorTest.php]: `: void`

Comment (Ocramius) [reply test/Generator/ValueGeneratorTest.php]: Please add type declarations to the parameters: drop them from the docblock

**Comment by Ocramius** [comment]
Check the failures, something is not working as expected: https://travis-ci.org/zendframework/zend-code/jobs/445104487#L653

**Comment by hxss** [comment]
lol. TABS > spaces
image

**Comment by Ocramius** [comment]
Last one: https://travis-ci.org/zendframework/zend-code/jobs/445367598#L675-L680

Maybe one level of nesting less would suffice :+1:

**Pull request merged**